### PR TITLE
fix #8485 ; make `default` public instead of redefined everywhere; use better implementation

### LINKED ISSUE #8485: `default` should be public instead of redefined everywhere
this should be exported (via `*`):
```nim
template default[T](t: typedesc[T]): T =
  var v: T
  v
```

instead of being redefined everywhere:

```nim
lib/core/seqs.nim:91:10:template default[T](t: typedesc[T]): T =
lib/pure/collections/queues.nim:157:10:template default[T](t: typedesc[T]): T =
lib/pure/collections/sets.nim:44:10:template default[T](t: typedesc[T]): T =
lib/pure/collections/sets.nim:284:10:template default[T](t: typedesc[T]): T =
lib/pure/collections/tableimpl.nim:119:10:template default[T](t: typedesc[T]): T =
tests/destructor/tcustomseqs.nim:101:10:template default[T](t: typedesc[T]): T =
tests/vm/tvmmisc.nim:16:8:  proc default[T](t: typedesc[T]): T {.inline.} = discard
```

also, it's useful in library & user code.

how about defining it (once) in `system.nim` ?

### PR DESCRIPTION
**Pull request by timotheecour**
* fixes #8485

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -4193,3 +4193,19 @@ when defined(genode):
       componentConstructHook(env)
         # Perform application initialization
         # and return to thread entrypoint.
+
+# needed pending https://github.com/nim-lang/Nim/issues/8486
+proc defaultImpl[T]():T=
+  var a: T
+  return a
```
Comment (hlaaftana) [inline lib/system.nim]: result variable instead?

Comment (timotheecour) [reply lib/system.nim]: done

**Comment by timotheecour** [comment]
@Araq PTAL tests green

**Comment by Araq** [comment]
Yeah but I don't like this workaround sorry, let's wait for https://github.com/nim-lang/Nim/issues/8486 to be fixed please.

**Comment by Araq** [comment]
This needs to be a language builtin.

**Pull request closed without merging**